code/blake2/Hacl.Impl.Blake2.Generic.fst

```diff
@@ -224,29 +224,38 @@ val gather_state: #a:Spec.alg -> #ms:m_spec -> st:state_p a ms -> m:block_w a ->
 		  (ensures (fun h0 _ h1 -> modifies (loc st) h0 h1 /\
 					state_v h1 st == Spec.gather_state a (as_seq h0 m) (v start)))
 
+inline_for_extraction noextract
+let get_sigma' (start: size_t { v start <= 144 }) (i: size_t { normalize (i <=. 15ul) }):
```

Comment:
why do you need to call `normalize` for `i` here? we can also use `+!` instead of `+.`, as `v start + v i < length sigmaTable`:
```
inline_for_extraction noextract
let get_sigma' (start: size_t { v start <= 144 }) (i: size_t {v i <= 15 }):
  Stack Spec.sigma_elt_t
    (requires (fun h -> True))
    (ensures  (fun h0 z h1 ->
      h0 == h1 /\ z == Lib.Sequence.(Spec.sigmaTable.[v start + v i])))
=
  get_sigma (start +! i)
```

Reply:
So this is a drive-by change because this proof kept failing repeatedly. I remember it was due to modulo proof operations, and writing out this helper made things work reliably. I added normalize because I knew that this would only ever be called with constant values for `i`, so that's one less thing to prove via SMT. (Or at least that was the intention!)

GitHub has a feature where you can suggest a change in the online interface, if you want to use that for `+!` vs `+.`. Otherwise, I'll fix it (eventually).

Thanks!

Reply:
That was mainly a question, not a suggestion :) I think using `+!` avoids those unnecessary proofs for modulo operations.

Reply:
Ah maybe that's why the proof was difficult in the first place. I didn't author this code so I thought +. was somewhat needed. Thanks for the review!